packages/lodestar/src/network/peers/metastore.ts

```diff
@@ -37,13 +41,23 @@ export class Libp2pPeerMetadataStore implements IPeerMetadataStore {
   constructor(config: IBeaconConfig, metabook: MetadataBook) {
     this.config = config;
     this.metabook = metabook;
-    this.encoding = this.typedStore("encoding", new StringType()) as PeerStoreBucket<ReqRespEncoding>;
-    this.metadata = this.typedStore("metadata", this.config.types.phase0.Metadata);
-    this.rpcScore = this.typedStore("score", this.config.types.Number64);
-    this.rpcScoreLastUpdate = this.typedStore("score-last-update", this.config.types.Number64);
+
+    const number64Serdes = typeToSerdes(this.config.types.Number64);
+    const metadataV2Serdes = typeToSerdes(this.config.types.altair.Metadata);
+    const stringSerdes: BucketSerdes<ReqRespEncoding> = {
+      serialize: (v) => Buffer.from(v, "utf8"),
+      deserialize: (b) => Buffer.from(b).toString("utf8") as ReqRespEncoding,
+    };
+
+    this.encoding = this.typedStore("encoding", stringSerdes);
+    // Discard existing `metadata` stored values. Store both phase0 and altair Metadata objects as altair
+    // Serializing altair.Metadata instead of phase0.Metadata has a cost of just `SYNC_COMMITTEE_SUBNET_COUNT // 8` bytes
+    this.metadata = this.typedStore("metadata-altair", metadataV2Serdes);
```

Comment:
The strategy to deal with the different Metadata types is to coerce `phase0.Metadata` into `altair.Metadata` with a zeroed syncnets value. Consumers that want the phase0 can just ignore the syncnets property. SSZ serialization works just fine since it ignores extra properties. Since metadata is ephemeral information the old `metadata` key is abandoned and not migrated.